File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn all-upper-case? [s]
  (when (some #(Character/isLetter %) s)
    (= s (str/upper-case s))))

(defn response-for [s]
  (cond
    (str/blank? s) "Fine. Be that way!"
    (all-upper-case? s) "Whoa, chill out!"
    (str/ends-with? s "?") "Sure."
    :else "Whatever."))

File: python/bob/bob.py
import re


def hey(phrase):
    phrase = re.sub('[\s+]', '', phrase)
    if phrase.isupper() and phrase.endswith('?'):
        return "Calm down, I know what I'm doing!"
    elif phrase.isupper():
        return "Whoa, chill out!"
    elif phrase.endswith('?'):
        return "Sure."
    elif not phrase:
        return "Fine. Be that way!"
    else:
        return "Whatever."
